src/main/java/de/tum/in/www1/artemis/web/rest/GradeStepResource.java

```diff
@@ -0,0 +1,159 @@
+
+    private final GradingScaleRepository gradingScaleRepository;
+
+    private final GradeStepRepository gradeStepRepository;
+
+    private final CourseRepository courseRepository;
+
+    public GradeStepResource(GradingScaleService gradingScaleService, GradingScaleRepository gradingScaleRepository, GradeStepRepository gradeStepRepository,
+            AuthorizationCheckService authCheckService, CourseRepository courseRepository) {
+        this.gradingScaleService = gradingScaleService;
+        this.gradingScaleRepository = gradingScaleRepository;
+        this.gradeStepRepository = gradeStepRepository;
+        this.authCheckService = authCheckService;
+        this.courseRepository = courseRepository;
+    }
+
+    /**
+     * GET /courses/{courseId}/grading-scale/grade-steps : Find all grade steps for the grading scale of a course
+     *
+     * @param courseId the course to which the grading scale belongs
+     * @return ResponseEntity with status 200 (Ok) with body a list of grade steps if the grading scale exists and 404 (Not found) otherwise
+     */
+    @GetMapping("/courses/{courseId}/grading-scale/grade-steps")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<List<GradeStep>> getAllGradeStepsForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get all grade steps for course: {}", courseId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        GradingScale gradingScale = gradingScaleRepository.findByCourseIdOrElseThrow(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        List<GradeStep> gradeSteps = gradeStepRepository.findByGradingScale_Id(gradingScale.getId());
+        return ResponseEntity.ok(gradeSteps);
+    }
+
+    /**
+     * GET /courses/{courseId}/exams/{examId}/grading-scale/grade-steps : Find all grade steps for the grading scale of an exam
+     *
+     * @param courseId the course to which the exam belongs
+     * @param examId the exam to which the grading scale belongs
+     * @return ResponseEntity with status 200 (Ok) with body a list of grade steps if the grading scale exists and 404 (Not found) otherwise
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/grading-scale/grade-steps")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<List<GradeStep>> getAllGradeStepsForExam(@PathVariable Long courseId, @PathVariable Long examId) {
+        log.debug("REST request to get all grade steps for exam: {}", examId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        GradingScale gradingScale = gradingScaleRepository.findByExamIdOrElseThrow(examId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        List<GradeStep> gradeSteps = gradeStepRepository.findByGradingScale_Id(gradingScale.getId());
+        return ResponseEntity.ok(gradeSteps);
+    }
```

Comment:
Note for the future: check if we need those or just get complete `GradingScale`s all the time (could save some code to maintain).